packages/connect/src/core/onCallFirmwareUpdate.ts

```diff
@@ -469,16 +470,19 @@ export const onCallFirmwareUpdate = async ({
 
     // We have completed binary download and we should notify sending an event,
     // if desktop wants to store it. We only do this for final FW, not intermediaries.
-    postMessage(
-        createUiMessage(UI.FIRMWARE_DOWNLOADED, {
+    const firmwareUpdateSource = DataManager.getSettings('firmwareUpdateSource');
+    const isCacheUsed = firmwareUpdateSource === 'production';
+    if (isCacheUsed) {
```

Comment:
We only emit the event to store the FW and releases JSON in cache while in production.